sdcm/sct_runner.py

```diff
@@ -62,185 +87,285 @@ def instance_root_disk_size(test_duration) -> int:
+
+        LOGGER.info("Installing required packages...")
+        login_user = self.LOGIN_USER
+        public_key = self.key_pair.public_key.decode()
+        result = remoter.sudo(shell_script_cmd(quote="'", cmd=f"""\
+            echo "fs.aio-max-nr = {AIO_MAX_NR_RECOMMENDED_VALUE}" >> /etc/sysctl.conf
+            echo "{login_user} soft nofile 4096" >> /etc/security/limits.conf
             echo "jenkins soft nofile 4096" >> /etc/security/limits.conf
             echo "root soft nofile 4096" >> /etc/security/limits.conf
-            sudo -u ubuntu mkdir -p /home/ubuntu/.ssh || true
-            echo "{self.key_pair().public_key.decode()}" >> /home/ubuntu/.ssh/authorized_keys
-            chmod 600 /home/ubuntu/.ssh/authorized_keys
-            mkdir -p -m 777 /home/ubuntu/sct-results
-            echo "cd ~/sct-results" >> /home/ubuntu/.bashrc
-            chown -R ubuntu:ubuntu /home/ubuntu/
-            apt clean
-            apt update
-            apt install -y python3-pip htop screen tree
+            sudo -u {login_user} mkdir -p /home/{login_user}/.ssh || true
+            echo "{public_key}" >> /home/{login_user}/.ssh/authorized_keys
+            chmod 600 /home/{login_user}/.ssh/authorized_keys
+            mkdir -p -m 777 /home/{login_user}/sct-results
+            echo "cd ~/sct-results" >> /home/{login_user}/.bashrc
+            chown -R {login_user}:{login_user} /home/{login_user}/
+            apt-get clean
+            apt-get update
+            apt-get install -y python3-pip htop screen tree
             pip3 install awscli
-            # docker
+
+            # Install Docker.
             apt-get install -y apt-transport-https ca-certificates curl gnupg-agent software-properties-common
-            curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+            curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
             apt-key fingerprint 0EBFCD88
             add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
-            apt update
-            apt install -y docker-ce docker-ce-cli containerd.io
-            usermod -aG docker {self.LOGIN_USER}
-            # add kubectl
+            apt-get update
+            apt-get install -y docker-ce docker-ce-cli containerd.io
+            usermod -aG docker {login_user}
+
+            # Install kubectl.
             curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
-            sudo install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
-            usermod -aG docker ubuntu || true
-            # configure Jenkins user
-            apt install -y openjdk-14-jre-headless
+            install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
+
+            # Configure Jenkins user.
+            apt-get install -y openjdk-11-jre-headless  # https://www.jenkins.io/doc/administration/requirements/java/
```

Comment:
SCT runner doesn't run any jenkins-related code. So, why Java version is changed here? I guess jenkins's supported versions is not the reason to change it here. Am I missing something?

Reply:
I discussed it with @bentsi and this is something that maybe we don't need at all on the SCT Runner.
We run _all_ our code inside a Docker container, so, for SCT it doesn't matter which version of JRE installed on the host system. I guess that there was an idea to add SCT runner as a dynamic Jenkins node, but not implemented.

The problem I solved here is absence of openjdk-14-jre-headless package in Ubuntu 20.04 repository.

@bentsi should we remove installation of JRE completely?

Reply:
let's keep it.

Reply:
@vponomaryov, FYI, the reason we're installing JRE here is to use the same image to deploy Jenkins nodes for SCT jobs.